dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/TrackedEntityAttributeController.java

```diff
@@ -120,35 +113,16 @@
         @RequestParam( required = false, defaultValue = "3" ) Integer expiration )
         throws WebMessageException
     {
-        TrackedEntityAttribute trackedEntityAttribute = trackedEntityAttributeService.getTrackedEntityAttribute( id );
-
-        if ( trackedEntityAttribute == null )
-        {
-            throw new WebMessageException( notFound( TrackedEntityAttribute.class, id ) );
-        }
-
         return reserve( id, 1, expiration ).get( 0 );
     }
 
     @GetMapping( "/{id}/requiredValues" )
     @ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
     public @ResponseBody Map<String, List<String>> getRequiredValues( @PathVariable String id )
-        throws WebMessageException
     {
         TrackedEntityAttribute trackedEntityAttribute = trackedEntityAttributeService.getTrackedEntityAttribute( id );
 
-        if ( trackedEntityAttribute == null )
```

Comment:
why are you not calling doing

`        TrackedEntityAttribute attribute = getTrackedEntityAttribute( id );`

instead of

`        TrackedEntityAttribute trackedEntityAttribute = trackedEntityAttributeService.getTrackedEntityAttribute( id );`

above? Since you remove the checks here